flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotProtocolTest.java

```diff
@@ -68,115 +52,79 @@
 
 public class SlotProtocolTest extends TestLogger {
 
-	private static TestingSerialRpcService testRpcService;
+	private static ScheduledExecutorService scheduledExecutorService;
 
 	@BeforeClass
 	public static void beforeClass() {
-		testRpcService = new TestingSerialRpcService();
+		scheduledExecutorService = new ScheduledThreadPoolExecutor(1);
 	}
 
 	@AfterClass
 	public static void afterClass() {
-		testRpcService.stopService();
-		testRpcService = null;
-	}
+		scheduledExecutorService.shutdown();
+
+		if (!scheduledExecutorService.isTerminated()) {
+			List<Runnable> runnables = scheduledExecutorService.shutdownNow();
 
-	@Before
-	public void beforeTest(){
-		testRpcService.clearGateways();
+			for (Runnable runnable : runnables) {
+				runnable.run();
```

Comment:
You're right. That way it does not make much sense. I think I wanted to call `scheduledExecutorService.awaitTermination` in the if condition in order to wait for the scheduled tasks to be executed.

Will replace it with `Executors.gracefulShutdown`.